systems/controllers/QPCommon.cpp

```diff
@@ -706,13 +726,14 @@ int setupAndSolveQP(NewQPControllerData *pdata, std::shared_ptr<drake::lcmt_qp_c
     for (int i=0; i<desired_body_accelerations.size(); i++) {
       if (desired_body_accelerations[i].weight > 0) {
         if (!inSupport(active_supports,desired_body_accelerations[i].body_id0)) {
-          pdata->r->forwardJac(desired_body_accelerations[i].body_id0,orig,1,Jb);
-          pdata->r->forwardJacDot(desired_body_accelerations[i].body_id0,orig,1,Jbdot);
+          pdata->r->forwardJac(desired_body_accelerations[i].body_id0,orig1,1,Jb);
+          auto Jdot_times_v = pdata->r->forwardJacDotTimesV(orig,desired_body_accelerations[i].body_id0,0,1,0);
+          Jbdotv = Jdot_times_v.value();
 
           for (int j=0; j<6; j++) {
             if (!std::isnan(desired_body_accelerations[i].body_vdot[j])) {
               pdata->Hqp += desired_body_accelerations[i].weight*(Jb.row(j)).transpose()*Jb.row(j);
-              f.head(nq) += desired_body_accelerations[i].weight*(robot_state.qd.transpose()*Jbdot.row(j).transpose() - desired_body_accelerations[i].body_vdot[j])*Jb.row(j).transpose();
+              f.head(nq) += desired_body_accelerations[i].weight*(Jbdotv(j) - desired_body_accelerations[i].body_vdot[j])*Jb.row(j).transpose();
```

Comment:
You could save a copy into a temporary here by using .noalias() on the left hand side.